### PR TITLE
http-client-java, fix duplicate Error model

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Main change is to avoid duplicating the Error model from Azure.Core.

The schema comes from 2 places.
One is from `Azure.Core.Foundations.Error` via typespec source.
Another is from the `PollOperationDetailsSchema` we created for mapping to Java azure-core class (on its `error` property).

They should be the same model, but here we created it twice. Due to a bug, in some test case, the deduplicate logic changed one of them to `Error1` (deduplicate should not apply to models in azure-core).

---

A minor change is to restore the partialupdate test. It was broken (partial-updated code not included in java file) when the module migrate to microsoft/typespec.

### CODE REVIEW

File: packages/http-client-java/emitter/src/external-schemas.ts [hunk 1]
```diff
@@ -26,36 +26,35 @@ export function createResponseErrorSchema(
...
+    language: {
+      default: {
+        namespace: "Azure.Core.Foundations",
+      },
+      java: {
+        namespace: "com.azure.core.models",
       },
     },
-  );
+  });
```
Comment (weidongxu-microsoft) [inline packages/http-client-java/emitter/src/external-schemas.ts:38]: This appears to be the bug, that java namespace is not set.

File: packages/http-client-java/emitter/src/code-model-builder.ts [hunk 2]
```diff
@@ -430,7 +430,15 @@ export class CodeModelBuilder {
     const deduplicateName = (schema: Schema) => {
       const name = schema.language.default.name;
       // skip models under "com.azure.core."
```
Comment (XiaofeiCao) [inline packages/http-client-java/emitter/src/code-model-builder.ts:432]: nit, update comment

File: packages/http-client-java/emitter/src/code-model-builder.ts [hunk 3]
```diff
@@ -430,7 +430,15 @@ export class CodeModelBuilder {
...
-      if (name && !schema.language.java?.namespace?.startsWith("com.azure.core.")) {
+      if (
+        name &&
+        // skip models under "com.azure.core."/"Azure.", if branded
+        !(
+          this.isBranded() &&
+          (schema.language.java?.namespace?.startsWith("com.azure.core.") ||
+            schema.language.default?.namespace?.startsWith("Azure."))
```
Comment (XiaofeiCao) [inline packages/http-client-java/emitter/src/code-model-builder.ts:439]: limit to Azure.Core as well?

**Review by XiaofeiCao — APPROVED** [review]

**Review by haolingdong-msft — APPROVED** [review]

File: packages/http-client-java/emitter/src/code-model-builder.ts (hunk 3 above)
Comment (weidongxu-microsoft) [reply packages/http-client-java/emitter/src/code-model-builder.ts:439]: I've got mind on TypeSpec model under `Azure.ResourceManager.` as well. Avoid touch them as well.

**Pull request merged**